### PR TITLE
LPS-143179 Migrate FDS headless display in `objects-web`

### PR DESCRIPTION
**Pull request by markocikos**
JIRA: https://issues.liferay.com/browse/LPS-143179

@liferay-objects In this PR, we are migrating `<clay:headless-data-set-display>` to `<frontend-data-set:headless-display>`. As a part of the migration, we are changing backend API from `frontend-taglib-clay` to API from `frontend-data-set` app. There is **no change in logic**, just a switch of tag and API classes, along with rename of variables to match the new API.

This is partly a followup on https://github.com/liferay-frontend/liferay-portal/pull/1722, as we are porting to `frontend-data-set` changes to API introduced in LPS-142886.

This is one of three parts of the migration:
- [x]  `search-experiences`
- [x] `objects`
- [ ] `commerce`

I tested changes on JSPs I could find in UI:
- [x]  view_object_definitions.jsp
- [x]  fields.jsp
- [x]  actions.jsp
- [x]  picklists > view.jsp
- [x]  edit_list_type_definition.jsp
- [x] view_object_entries
- [x] relationships
- [x] layouts

Please check the rest!

/cc @LuismiBarcos @javierdearcos

### CODE REVIEW

File: modules/apps/frontend-data-set/frontend-data-set-api/src/main/java/com/liferay/frontend/data/set/view/table/DateTimeFDSTableSchemaField.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+package com.liferay.frontend.data.set.view.table;
+
+/**
+ * @author Guilherme Camacho
+ */
+public class DateTimeFDSTableSchemaField extends BaseDateFDSTableSchemaField {
+
+	@Override
+	public String getContentRenderer() {
+		return "datetime";
```
Comment (guilhermedcamacho) [inline modules/apps/frontend-data-set/frontend-data-set-api/src/main/java/com/liferay/frontend/data/set/view/table/DateTimeFDSTableSchemaField.java:24]: Hey @markocikos , we had to change it to "dateTime" as requested by Brian. He merged it some hours ago [here](https://github.com/brianchandotcom/liferay-portal/pull/111367), please do a rebase when synced upstream. In the meantime @marcelabc or @gabrielwas can review the other parts of your code. Thanks a lot!

File: modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/frontend/data/set/view/table/ObjectDefinitionsTableFDSView.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+		FDSTableSchemaField systemFDSTableSchemaField =
+			fdsTableSchemaBuilder.addFDSTableSchemaField("system", "system");
+
+		systemFDSTableSchemaField.setContentRenderer("boolean");
+
+		FDSTableSchemaField dateFDSTableSchemaField =
+			fdsTableSchemaBuilder.addFDSTableSchemaField(
+				"dateModified", "modified-date");
+
+		dateFDSTableSchemaField.setContentRenderer("date");
```
Comment (guilhermedcamacho) [inline modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/frontend/data/set/view/table/ObjectDefinitionsTableFDSView.java:61]: Hey @markocikos , we had to change from "date" to "dateTime" here. Brian merged it some hours ago in the same [PR](https://github.com/brianchandotcom/liferay-portal/pull/111367) I've mentioned in the other comment, please do a rebase when synced upstream. In the meantime @marcelabc or @gabrielwas can review the other parts of your code. Thanks a lot!

**Review by guilhermedcamacho — CHANGES_REQUESTED** [review]

**Comment by markocikos** [comment]
@guilhermedcamacho Rebased, fixed a conflict, added requested changes in https://github.com/liferay-objects/liferay-portal/pull/298/commits/39f075508c6a4ca355cd87212a3da86aad4e6134, retested usages, force-pushed.

**Comment by guilhermedcamacho** [comment]
ci:test:relevant

**Comment by gabrielwas** [comment]
Just started reviewing :)

**Comment by gabrielwas** [comment]
Rebased and resent in https://github.com/liferay-objects/liferay-portal/pull/321

**Pull request closed without merging**